Fix unit tests on macOS and Linux (where the default line ending is LF)

0xced opened the pull request.

JoshClose commented:
Could you resolve the conflicts for this? I just tried to merge and there are a lot of them.

0xced commented:
I just rebased on the master branch and force-pushed.

I also fixed the `IParserConfiguration` and `IWriterConfiguration` documentation about the default NewLine which is now CRLF.

JoshClose commented:
Thanks!
